MsixCore/msixmgr/msixmgr.cpp

```diff
@@ -246,62 +297,251 @@ int main(int argc, char * argv[])
+                    //if (FAILED(hr))
+                    //{
+                    //    std::wcout << std::endl;
+                    //    std::wcout << "Failed to create and mount vhd with error: " << hr << std::endl;
+                    //    std::wcout << std::endl;
+                    //    return hr;
+                    //}
+                    //else
+                    //{
+                    //    std::wcout << std::endl;
+                    //    std::wcout << "Successfully created VHD and mounted to drive: " << mountPoint.c_str() << std::endl;
+                    //    std::wcout << std::endl;
+                    //}
+
+                    return ERROR_NOT_SUPPORTED;
+                }
+                else
+                {
+                    std::vector<std::wstring> skippedFiles;
+                    std::vector<std::wstring> failedPackages;
+                    std::vector<HRESULT> failedPackagesErrors;
+                    RETURN_IF_FAILED(MsixCoreLib::Unpack(
+                        packageSourcePath,
+                        unpackDestination,
+                        cli.IsApplyACLs(),
+                        cli.IsValidateSignature(),
+                        skippedFiles,
+                        failedPackages,
+                        failedPackagesErrors));
+
+                    std::wcout << std::endl;
+                    std::wcout << "Finished unpacking packages to: " << unpackDestination << std::endl;
+                    std::wcout << std::endl;
+
+                    OutputUnpackFailures(packageSourcePath, skippedFiles, failedPackages, failedPackagesErrors);
+                }
             }
-            else
+            return hr;
+        }
+        case OperationType::ApplyACLs:
+        {
+            std::vector<std::wstring> packageFolders;
+            packageFolders.push_back(cli.GetPackageFilePathToInstall()); // we're not actually installing anything. The API just returns the file path name we need.
+            RETURN_IF_FAILED(MsixCoreLib::ApplyACLs(packageFolders));
+            return S_OK;
+        }
+        case OperationType::MountImage:
+        {
+            if (cli.GetFileType() == WVDFileType::CIM)
             {
+                if (cli.GetVolumeId().empty())
+                {
+                    std::wcout << std::endl;
+                    std::wcout << "Please provide a volume id in order to mount a CIM image" << std::endl;
+                    std::wcout << std::endl;
+                    return E_INVALIDARG;
+                }
+
+                std::wstring volumeIdString = cli.GetVolumeId();
+                GUID volumeIdFromString;
+                if (UuidFromStringW((RPC_WSTR)(cli.GetVolumeId().c_str()), &volumeIdFromString) != RPC_S_OK)
+                {
+                    std::wcout << std::endl;
+                    std::wcout << "Failed to convert specified volume id {" << volumeIdString << "} to GUID" << std::endl;

```

Comment:
>volumeIdString [](start = 79, length = 14)

can this already have curly braces?

Reply:
The expected workflow is that the user would provide a GUID _without_ the curly braces.